### ISSUE TITLE
Version 2.3.0 no longer works when magento/module-instant-purchase is not availabe

### ISSUE THREAD

**Issue by hostep** [mollie/magento2#470]
**Describe the bug**
Version 2.3.0 of this module has a dependency on Magento's InstantPurchase module, but we don't have it installed and thus `setup:di:compile` breaks now.

**Used versions**
- *Magento Version number(eg 2.3.5)*: 2.4.3-p1
- *Open source/Enterprise/B2b*: OS
- *Mollie version number (Check configuration)*: 2.3.0

**To Reproduce**
1. Have a vanilla Magento installation
2. Edit the `composer.json` file and add:
```
    "replace": {
        "magento/module-instant-purchase": "*",
    }
```
3. Run `composer update`
4. Run `bin/magento setup:di:compile`

**Expected behavior**
No error

**Actual behavior**
Error:
```
Interface 'Magento\InstantPurchase\PaymentMethodIntegration\AvailabilityCheckerInterface' not found#0 vendor/composer/ClassLoader.php(571): include()
#1 vendor/composer/ClassLoader.php(428): Composer\Autoload\includeFile('...')
#2 [internal function]: Composer\Autoload\ClassLoader->loadClass('Mollie\\Payment\\...')
#3 [internal function]: spl_autoload_call('Mollie\\Payment\\...')
#4 setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php(134): class_exists('Mollie\\Payment\\...')
#5 setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php(117): Magento\Setup\Module\Di\Code\Reader\ClassesScanner->includeClass('Mollie\\Payment\\...', '...')
...
```

**Screenshots**

**Additional context**
Not sure if this is an actual bug, but we don't need the InstantPurchase module on a particular shop, so we chose not to install it (which is a [common practice](https://www.integer-net.com/make-magento-2-small-again/) in the Magento world these days).
Upon updating mollie to the latest version (2.3.0), we notice that it now has a dependency on this module and `setup:di:compile` breaks.

Is there a way to make your module not depend on this module?
An option would be to split up your module in different smaller modules, like how they do in the [MultiSafepay composer.json file](https://github.com/MultiSafepay/magento2/blob/9efb40d1d8102096c4cb8d82695a840d022bb8fb/composer.json#L6-L10) for example, where the module that is responsible for anything related to MSI is a separate module, which we can easily not install when we don't need the MSI module.
A similar solution could be used over here for the InstantPurchase module?

Is not a huge problem in my opinion, it's just something unexpected that I didn't see coming with the update of this module.

This issue is slightly related to https://github.com/mollie/magento2/issues/459, even though adding a dependency definition for all required magento modules (which I can definitely recommend!) wouldn't solve our particular case, since the dependency resolution of composer ignores the `replace` section.

**Comment by Frank-Magmodules** [comment]
HI @hostep ,

Thank you for opening this issue, we are happy to share that we've just released the new [2.5.0](https://github.com/mollie/magento2/releases/tag/v2.5.0) version where we covered this issue. Thank you for the detailed issue report. 

We are closing this issue now but please feel free to reopen the issue if this still occurs.

**Comment by hostep** [comment]
Thanks @Frank-Magmodules!

Interesting [trick](https://github.com/mollie/magento2/commit/49466928b1d526c8593b57b0d2c7c741f2202f1a#diff-826b255d949abf19002a1fc20d1cdc95ee5fc68c6b871409e6e7cf14147682f8)!
I was almost convinced this would trip over composer 2's optimised autoloader due to the incorrect namespace of `Magento` vs `Mollie`, but it doesn't. So I guess it will work fine! (But I haven't tested it out yet in practice.)
